backends/tc/backend.cpp

```diff
@@ -122,15 +121,34 @@ void Backend::serialize() const {
             outstream->flush();
         }
     }
+    auto progName = options.file;
+    auto filename = progName.findlast('/');
+    if (filename) progName = filename;
+    progName = progName.exceptLast(3);
+    progName = progName.trim("/\t\n\r");
+    cstring parserFile = progName + "_parser.c";
+    cstring postParserFile = progName + "_post_parser.c";
+    cstring headerFile = progName + "_parser.h";
     if (!options.cFile.isNullOrEmpty()) {
-        auto cstream = openFile(options.cFile, false);
-        if (cstream == nullptr) return;
-        if (ebpf_program == nullptr) return;
-        EBPF::CodeBuilder c(target);
-        ebpf_program->emit(&c);
-        *cstream << c.toString();
-        cstream->flush();
+        parserFile = options.cFile + parserFile;
```

Comment:
when -c is specified , shouldn't it be options.cFile + "_parser.c"; etc instead of adding source name also as prefix